File: src/2025-03-04-03-28-40--deepseek-chat-0304/javascript/exercises/practice/variable-length-quantity/variable-length-quantity.js
export const encode = (numbers) => {
  if (!numbers || numbers.length === 0) return [];
  
  const result = [];
  
  for (const num of numbers) {
    if (num < 0 || num > 0xFFFFFFFF) {
      throw new Error('Invalid number. Must be 32-bit unsigned integer');
    }
    
    let value = num;
    let bytes = [];
    
    do {
      let byte = value & 0x7F;
      value >>>= 7;
      if (bytes.length > 0) {
        byte |= 0x80;
      }
      bytes.unshift(byte);
    } while (value > 0);
    
    result.push(...bytes);
  }
  
  return result;
};

export const decode = (bytes) => {
  if (!bytes || bytes.length === 0) return [];
  
  const result = [];
  let num = 0;
  
  for (let i = 0; i < bytes.length; i++) {
    const byte = bytes[i];
    
    if ((byte & 0x80) === 0) {
      // Last byte in sequence
      num = (num << 7) | (byte >>> 0);
      result.push(num);
      num = 0;
    } else {
      // More bytes to come
      num = (num << 7) | ((byte & 0x7F) >>> 0);
      
      // Check for incomplete sequence
      if (i === bytes.length - 1) {
        throw new Error('Incomplete sequence');
      }
    }
  }
  
  return result;
};

File: src/2025-03-04-03-28-40--deepseek-chat-0304/python/exercises/practice/variable-length-quantity/variable_length_quantity.py
def encode(numbers):
    """Encode a list of numbers into VLQ bytes."""
    result = []
    for number in numbers:
        if number < 0:
            raise ValueError("Only unsigned integers are supported")
        if number > 0xFFFFFFFF:
            raise ValueError("Number too large for 32-bit VLQ")
            
        # Handle zero case
        if number == 0:
            result.append(0)
            continue
            
        # Encode the number
        bytes_ = []
        while number > 0:
            byte = number & 0x7F
            number >>= 7
            if bytes_:
                byte |= 0x80  # Set continuation bit
            bytes_.insert(0, byte)
        result.extend(bytes_)
    return result


def decode(bytes_):
    """Decode VLQ bytes into a list of numbers."""
    if not bytes_:
        return []
        
    numbers = []
    number = 0
    for i, byte in enumerate(bytes_):
        # Check for incomplete sequence
        if i == len(bytes_) - 1 and (byte & 0x80):
            raise ValueError("incomplete sequence")
            
        number = (number << 7) | (byte & 0x7F)
        if not (byte & 0x80):
            numbers.append(number)
            number = 0
    return numbers
